src/Microsoft.Framework.Runtime.Hosting/Json/JsonObject.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Framework.Runtime.Json
+{
+    internal class JsonObject : JsonValue
+    {
+        private readonly IDictionary<string, JsonValue> _data;
+
+        public JsonObject(IDictionary<string, JsonValue> data, JsonPosition position)
+            : base(position)
+        {
+            if (data == null)
+            {
+                throw new ArgumentNullException(nameof(data));
+            }
+
+            _data = data;
+        }
+
+        public ICollection<string> Keys
+        {
+            get { return _data.Keys; }
+        }
+
+        public JsonValue Value(string key)
+        {
+            JsonValue result;
+            if (!_data.TryGetValue(key, out result))
+            {
+                result = null;
+            }
+
+            return result;
+        }
+
+        public T ValueAs<T>(string key, Func<JsonValue, T> cast)
+        {
+            return cast(Value(key));
+        }
+
+        public JsonObject ValueAsJsonObject(string key)
+        {
+            return ValueAs<JsonObject>(key, value =>
+            {
+                return value as JsonObject;
+            });
+        }
+
+        public JsonString ValueAsString(string key)
+        {
+            return ValueAs(key, value => value as JsonString);
+        }
+
+        public bool ValueAsBoolean(string key, bool defaultValue = false)
+        {
+            return ValueAs(key, value =>
+            {
+                if (value != null && value is JsonBoolean)
```

Comment:
condition could be simplified:
a) `as` cast and check on null
b) remove null check because if `value` is null `is` will return false